plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java

```diff
@@ -910,6 +927,375 @@ public void testSettingChangeDataFeedEnabledProperty()
+        assertUpdate("INSERT INTO " + tableName + " VALUES('url2', 'domain2', 2)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES('url3', 'domain3', 3)", 1);
+        MaterializedResult result = computeActual("SELECT * FROM TABLE(system.table_changes('test_schema', '" + tableName + "', 0))");
+        List<MaterializedRow> resultWithoutTimestamp = getCdfColumnsWithoutTimestamp(result);
+        assertThat(resultWithoutTimestamp).containsOnly(
+                row("url1", "domain1", 1, "insert", 1),
+                row("url2", "domain2", 2, "insert", 2),
+                row("url3", "domain3", 3, "insert", 3));
+
+        assertUpdate("UPDATE " + tableName + " SET page_url = 'url22' WHERE domain = 'domain2'", 1);
+        assertQuerySucceeds("ALTER TABLE " + tableName + " SET PROPERTIES change_data_feed_enabled = true");
+        assertUpdate("UPDATE " + tableName + " SET page_url = 'url33' WHERE views = 3", 1);
+        assertUpdate("DELETE FROM " + tableName + " WHERE page_url = 'url1'", 1);
+
+        assertQuery("SELECT * FROM " + tableName, "VALUES" +
+                "('url22', 'domain2', 2), " +
+                "('url33', 'domain3', 3)");
+
+        assertQueryFails("SELECT * FROM TABLE(system.table_changes('test_schema', '" + tableName + "', 3))",
+                "Change Data Feed is not enabled for the requested versions range");
+        assertQueryFails("SELECT * FROM TABLE(system.table_changes('test_schema', '" + tableName + "'))",
+                "Change Data Feed is not enabled for the requested versions range");
+        result = computeActual("SELECT * FROM TABLE(system.table_changes('test_schema', '" + tableName + "', 5))");
+        resultWithoutTimestamp = getCdfColumnsWithoutTimestamp(result);
+        assertThat(resultWithoutTimestamp).containsOnly(
+                row("url3", "domain3", 3, "update_preimage", 6),
+                row("url33", "domain3", 3, "update_postimage", 6),
+                row("url1", "domain1", 1, "delete", 7));
+    }
+
+    @Test
+    public void testReadingChangesFromCTASTableCdfEnabled()
+    {
+        String tableName = "test_basic_operations_on_table_with_cdf_enabled_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " WITH (change_data_feed_enabled = true) AS SELECT * FROM (VALUES" +
+                "('url1', 'domain1', 1), " +
+                "('url2', 'domain2', 2)) t(page_url, domain, views)",
+                2);
+
+        MaterializedResult result = computeActual("SELECT * FROM TABLE(system.table_changes('test_schema', '" + tableName + "'))");
+        List<MaterializedRow> resultWithoutTimestamp = getCdfColumnsWithoutTimestamp(result);
+        assertThat(resultWithoutTimestamp).containsOnly(
+                row("url1", "domain1", 1, "insert", 0),
+                row("url2", "domain2", 2, "insert", 0));
+    }
+
+    @Test
+    public void testVacuumCorrectlyDeletesCdfFiles()
```

Comment:
testVacuumCdfFiles

"Correctly" is redundant; tests most often test that something is correct